### PR TITLE
[windows] some fs.watch fixes

### PR DESCRIPTION
**Pull request by cirospaciari**
### What does this PR do?

Fix: https://github.com/oven-sh/bun/issues/8361

- [ ] Documentation or TypeScript types (it's okay to leave the rest blank in this case)
- [X] Code changes

### How did you verify your code works?

Existing Tests

### CODE REVIEW

File: test/js/bun/spawn/spawn.test.ts [hunk 1]
```diff
@@ -94,6 +94,7 @@ for (let [gcTick, label] of [
         gcTick();
       });
 
+<<<<<<< Updated upstream
```
Comment (paperclover) [inline test/js/bun/spawn/spawn.test.ts]: lol merge conflict committed

Comment (cirospaciari) [reply test/js/bun/spawn/spawn.test.ts]: lol will revert this file, should not be here

**Comment by Jarred-Sumner** [comment]
looks like the windows tests aren't there yet

[Link to file](https://github.com/oven-sh/bun/blob/dbb28fc7c7ada2a0550c20b326f85eaf57b7c71d/test%5Cjs%5Cnode%5Chttp2%5Cnode-http2.test.js)


```ts
test\js\node\http2\node-http2.test.js

// @known-failing-on-windows: panic "integer overflow"
bun test v1.0.26-canary.16 (dbb28fc7)




1 | export default "node";
        ^
error: Session closed with error code 9
 errno: 9
  code: "ERR_HTTP2_SESSION_ERROR"

      at node:1:5
      at goaway (node:1:686)
      at read (native:1:1)
      at #onRead (node:1:711)
      at emit (native:1:1)
      at addChunk (node:1:1940)
      at readableAddChunk (node:1:1894)
      at data (node:1:52)
✗ Client Basics > should be able to send a GET request [384.65ms]

uh-oh: integer overflow
bun will crash now 😭😭😭

----- bun meta -----
Bun v1.0.26-canary.16 (dbb28fc7) Windows x64 
TestCommand: bunfig 
Elapsed: 743ms | User: 109ms | Sys: 15ms
RSS: 107.38MB | Peak: 107.38MB | Commit: 0.15GB | Faults: 26470
----- bun meta -----

Search GitHub issues https://bun.sh/issues or join in #windows channel in https://bun.sh/discord

thread 3388 panic: integer overflow
???:?:?: 0x7ff79435848d in ??? (bun.exe)
???:?:?: 0x7ff79423090c in ??? (bun.exe)
???:?:?: 0x7ff79402ca31 in ??? (bun.exe)
???:?:?: 0x7ff796086e7a in ??? (bun.exe)
???:?:?: 0x2bd8000115d in ??? (???)
???:?:?: 0xc04cfb41a984 in ??? (???)
```

File: src/js/node/http2.ts [hunk 2]
```diff
@@ -874,12 +877,13 @@ class ClientHttp2Session extends Http2Session {
...
-        self.#queue.push(buffer);
-      } else {
+      if (!socket) return;
+      if (self.#connected) {
         // redirect writes to socket
         socket.write(buffer);
+      } else {
+        //queue
+        self.#queue.push(buffer);
       }
```
Comment (necmettindev) [inline src/js/node/http2.ts]: if(self[bunHTTP2Socket])

Comment (cirospaciari) [reply src/js/node/http2.ts]: The socket can be attached and not connected/secureConnected yet

**Comment by cirospaciari** [comment]
Closed in favor of https://github.com/oven-sh/bun/pull/8746

**Pull request closed without merging**